arch/arm/boot/dts/broadcom/rp1.dtsi

```diff
@@ -1109,16 +1109,17 @@
 
 			interrupts = <RP1_INT_MIPI0 IRQ_TYPE_LEVEL_HIGH>;
 
-			clocks = <&rp1_clocks RP1_CLK_MIPI0_CFG>,  // required, config bus clock
-				 <&rp1_clocks RP1_CLK_MIPI0_DPI>,  // required, pixel clock
-				 <&clksrc_mipi0_dsi_byteclk>,    // internal, parent for divide
-				 <&clk_xosc>;                    // hardwired to DSI "refclk"
-			clock-names = "cfgclk", "dpiclk", "byteclk", "refclk";
+			clocks = <&rp1_clocks RP1_CLK_MIPI0_CFG>,
+				 <&rp1_clocks RP1_CLK_MIPI0_DPI>,
+				 <&rp1_clocks RP1_CLK_MIPI0_DSI_BYTECLOCK>,
+				 <&clk_xosc>,                // hardwired to DSI "refclk"
+				 <&rp1_clocks RP1_PLL_SYS>;  // alternate parent for divide
+			clock-names = "cfgclk", "dpiclk", "byteclk", "refclk", "pllsys";
 
 			assigned-clocks = <&rp1_clocks RP1_CLK_MIPI0_CFG>,
-					  <&rp1_clocks RP1_CLK_MIPI0_DPI>;
+					  <&rp1_clocks RP1_CLK_MIPI0_DPI>,
+					  <&rp1_clocks RP1_CLK_MIPI0_DSI_BYTECLOCK>;
 			assigned-clock-rates = <25000000>;
```

Comment:
What are the semantics of having fewer assigned-clock-rates than assigned-clocks. Or, to put it another way, what is the effect of assigning a clock without giving it a rate?

Reply:
Going by the documentation, it probably does nothing. (I imagined it might give the driver some kind of exclusive right to set the rate and parent at run-time, but can find no evidence for this). I'd better try it with this removed.

Reply:
Yes it seems to work just as well without the last two assigned clocks. I will remove them and force the change into the penultimate commit.